Textfield/Textfield.jsx

```diff
@@ -119,24 +120,13 @@ class Textfield extends Component {
   }
 
   componentDidMount() {
-    this._setInvalid(defaultProps, this.props, this.MDComponent);
     this.setState({
       showFloatingLabel: true
     });
   }
 
-  _setInvalid(oldprops, newprops, textfield) {
-    if (
-      "valid" in oldprops &&
-      "valid" in newprops &&
-      oldprops.valid !== newprops.valid
-    ) {
-      textfield.valid = newprops.valid;
-    }
-  }
-
   componentWillUpdate(nextProps) {
-    this._setInvalid(this.props, nextProps, this.MDComponent);
+    setValid(this.props, nextProps, this.MDComponent);
```

Comment:
This class doesn't have any `MDComponent`, `TextfieldInput` has. Shouldn't this be there?

Reply:
Yes, that is correct